quantmsrescore/ms2_model_manager.py

```
@@ -33,20 +36,53 @@ def __init__(self,
         return self.model_str
 
+    def _download_models(self, model_zip_file_path: str, overwrite: bool = True) -> None:
+        """Download models if not done yet."""
+        url = self.model_url
+        parsed = urllib.parse.urlparse(url)
+        if parsed.scheme not in ("http", "https"):
+            raise ValueError(f"Disallowed URL scheme: {parsed.scheme}")
+
+        if not os.path.exists(model_zip_file_path):
+            if not overwrite and os.path.exists(model_zip_file_path):
+                raise FileExistsError(f"Model file already exists: {model_zip_file_path}")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)